core/src/OpenMP/Kokkos_OpenMP_Parallel.hpp

```diff
@@ -72,6 +72,11 @@
 namespace Kokkos {
 namespace Impl {
 
+inline bool in_serial() {
+  return (OpenMP::in_parallel() &&
+          !(omp_get_nested() && (omp_get_level() == 1)));
+}
```

Comment:
The problem is not that it's nested. It's that you want to execute the code in parallel but the loop require nested parallelism which was not enable. I don't think that it should be called `is_nested` because that's not the issue itself. I can change it to `execute_in_serial` if you prefer.